include/wx/typeinfo.h

```diff
@@ -24,16 +24,6 @@
 
 #include "wx/defs.h"
 
-//
-// Left for compatibility reasons
-//
-
-#define _WX_DECLARE_TYPEINFO_CUSTOM(CLS, IDENTFUNC)
-#define WX_DECLARE_TYPEINFO_INLINE(CLS)
-#define WX_DECLARE_TYPEINFO(CLS)
-#define WX_DEFINE_TYPEINFO(CLS)
-#define WX_DECLARE_ABSTRACT_TYPEINFO(CLS)
-
```

Comment:
I think we should still leave those just in case people use it in their own code, even if they're not used by wx itself any longer. It doesn't cost much to do it and people appreciate when their code continues to compile with newer wx versions without errors.

Reply:
I went off the comments regarding the macros existing for internal use only but I'll make a few changes and restore the missing macros to make it backward compatible.